src/mixins/yield-erc/ERC20Vault.sol

```diff
@@ -0,0 +1,149 @@
+    function deposit(address to, uint256 underlyingAmount) external virtual returns (uint256 shares) {
+        shares = underlyingAmount.fdiv(exchangeRate(), baseUnit);
+        // Determine the equivalent amount of shares and mint them.
+        _mint(to, shares);
+
+        emit Deposit(to, underlyingAmount);
+
+        // Transfer in underlying tokens from the user.
+        // This will revert if the user does not have the amount specified.
+        underlying.safeTransferFrom(msg.sender, address(this), underlyingAmount);
+
+        afterDeposit(underlyingAmount);
+    }
+
+    /// @notice Withdraw a specific amount of underlying tokens.
+    /// @param to The address to receive underlying tokens corresponding to the withdrawal.
+    /// @param underlyingAmount The amount of underlying tokens to withdraw.
+    function withdraw(address to, uint256 underlyingAmount) external virtual returns (uint256 shares) {
+        shares = underlyingAmount.fdiv(exchangeRate(), baseUnit);
+
+        // Determine the equivalent amount of shares and burn them.
+        // This will revert if the user does not have enough shares.
+        _burn(msg.sender, shares);
+
+        emit Withdraw(msg.sender, to, underlyingAmount);
+
+        // Withdraw from strategies if needed and transfer.
+        beforeWithdraw(underlyingAmount);
+
+        underlying.safeTransfer(to, underlyingAmount);
+    }
+
+    /// @notice Redeem a specific amount of shares for underlying tokens.
+    /// @param to The address to receive underlying tokens corresponding to the withdrawal.
+    /// @param shareAmount The amount of shares to redeem for underlying tokens.
+    function redeem(address to, uint256 shareAmount) external virtual returns (uint256 underlyingAmount) {
+
+        // Determine the equivalent amount of underlying tokens.
+        underlyingAmount = shareAmount.fmul(exchangeRate(), baseUnit);
+
+        // Burn the provided amount of shares.
+        // This will revert if the user does not have enough shares.
+        _burn(msg.sender, shareAmount);
+
+        emit Withdraw(msg.sender, to, underlyingAmount);
+
+        // Withdraw from strategies if needed and transfer.
+        beforeWithdraw(underlyingAmount);
+
+        underlying.safeTransfer(to, underlyingAmount);
+    }
+
+    function beforeWithdraw(uint256 underlyingAmount) internal virtual {}
+
+    function afterDeposit(uint256 underlyingAmount) internal virtual {}
+
+    /*///////////////////////////////////////////////////////////////
+                        VAULT ACCOUNTING LOGIC
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Returns a user's Vault balance in underlying tokens.
+    /// @param user The user to get the underlying balance of.
+    /// @return The user's Vault balance in underlying tokens.
+    function balanceOfUnderlying(address user) external view returns (uint256) {
+        return balanceOf[user].fmul(exchangeRate(), baseUnit);
+    }
```

Comment:
Same as above, here you should round up.

Reply:
good point